### PR TITLE
#323 Fix for missed TINY_IMAGE during DockerClient initialization

### PR DESCRIPTION
**Pull request by gorelikov**
Fix for `No such image: alpine:3.5 #323`
@bsideup

### CODE REVIEW

File: core/src/main/java/org/testcontainers/DockerClientFactory.java [hunk 1]
```diff
@@ -98,11 +98,8 @@ public DockerClient client() {
 
             checkVersion(version.getVersion());
 
-            List<Image> images = client.listImagesCmd().exec();
             // Pull the image we use to perform some checks
-            if (images.stream().noneMatch(it -> it.getRepoTags() != null && asList(it.getRepoTags()).contains(TINY_IMAGE))) {
-                client.pullImageCmd(TINY_IMAGE).exec(new PullImageResultCallback()).awaitSuccess();
-            }
+            checkAndPullImage(client, TINY_IMAGE);
```
Comment (bsideup) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: `checkDiskSpaceAndHandleExceptions` calls `checkDiskSpace` which calls `runInsideDocker` which calls `checkAndPullImage` after your change.

It means that we could remove `checkAndPullImage` from here and inline `checkAndPullImage` in `runInsideDocker`.

WDYT?

File: core/src/test/java/org/testcontainers/DockerClientFactoryTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Created mgorelikov on 05/04/2017
+ * Test for {@link DockerClientFactory}.
+ */
+public class DockerClientFactoryTest {
+
+  private static final String SUCCESS = "SUCCESS";
+
+  @Test
+  public void runCommandInsideDockerShouldPullImageIfItDoesNotExistsLocally() {
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/DockerClientFactoryTest.java]: I think this test does too much. What it should test that `runInsideDocker` does not fail if image wasn't pulled yet

File: core/src/test/java/org/testcontainers/DockerClientFactoryTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+import org.junit.Test;
+import org.testcontainers.dockerclient.LogToStringContainerCallback;
+import org.testcontainers.utility.TestcontainersConfiguration;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Created mgorelikov on 05/04/2017
```
Comment (bsideup) [inline core/src/test/java/org/testcontainers/DockerClientFactoryTest.java]: auto-generated doc comment. I would remove it, doesn't help here.

However, you can include yourself in contributors list :)

Comment (gorelikov) [inline core/src/test/java/org/testcontainers/DockerClientFactoryTest.java]: Fixed. Sorry for that

File: core/src/main/java/org/testcontainers/DockerClientFactory.java (hunk 1 above)
Comment (gorelikov) [inline core/src/main/java/org/testcontainers/DockerClientFactory.java]: I removed that call, but kept the method checkAndPullImage. I don't like long methods, so i would keep the separate method, if you don't mind :)

File: core/src/test/java/org/testcontainers/DockerClientFactoryTest.java (hunk 2 above)
Comment (gorelikov) [inline core/src/test/java/org/testcontainers/DockerClientFactoryTest.java]: You are right, I kept here only `runInsideDocker` method call

**Comment by bsideup** [comment]
@gorelikov merged, thanks for your contribution! :)

**Comment by dddpaul** [comment]
Hi there. It's crucial PR for our CI because we still cannot test and build up services with Jenkins. I'm afraid it's to long to wait for 1.3.0 release. So I'd be glad if it would be rescheduled to 1.2.1, for example :)

**Comment by kiview** [comment]
You could always fallback to using a Jitpack dependency on the merged PR:
https://www.testcontainers.org/usage.html#maven-dependencies

**Pull request merged**